ansible/ansible-ai-connect-service
# AAP-29426: Automate two reports and save output in csv files

Jira Issue: https://issues.redhat.com/browse/AAP-29426

## Description
This PR automates production of the "Trial User reports".

It adds a Django admin command that is triggered by a k8s `CronJob` [in](https://github.com/ansible/ansible-wisdom-ops/pull/950) `ansible-wisdom-ops`.

The command extracts the data and builds CSV response. It is sent to a configurable "postman".

I have implemented "postmen" for Slack Web Hooks, Slack WebAPI and GDrive.

Slack Web Hooks do not support attachments; so the report is inline in the Slack message.

![image](https://github.com/user-attachments/assets/bcb66c77-39ef-47a3-b6a7-cd465b959b96)

## Testing
- Create a namespace somewhere.
- Deploy `Secret`'s[*] `wisdom-service`, `wisdom-service-deployment` and `wca-awssm` to the namespace
- Deploy `ConfigMap`[*] `wisdom-service` to the namespace.
- Deploy https://github.com/ansible/ansible-wisdom-ops/pull/950 to the namespace; with the following changes:-
  - Change `schedule` to `*/5 * * * *` (every 5 minutes for testing)
  - Additional environment variables:
```
env:
  - name: ANSIBLE_AI_ONE_CLICK_REPORTS_POSTMAN
    value: "slack-webhook"
  - name: ANSIBLE_AI_ONE_CLICK_REPORTS_CONFIG
    value: '{"slack-webhook-url": <I can advise>}'
```              
  - Change `image` to `image: quay.io/ansible/ansible-ai-connect-service:pr-1260-ddc4c663ab1e05943f7fe62e800d844130db2761`

You should notice posts to the configured Slack channel every minutes.

[*] The content of these resources can be copied from `stage2-west`, `wisdom-service-staging` namespace.

### Scenarios tested
As above.

## Production deployment
https://github.com/ansible/ansible-wisdom-ops/pull/950
